tools/@aws-cdk/spec2cdk/lib/cdk/relationship-decider.ts

```diff
@@ -1,69 +1,67 @@
 import { Property, RelationshipRef, Resource, RichProperty, SpecDatabase } from '@aws-cdk/service-spec-types';
+import { Module, SelectiveModuleImport, Type } from '@cdklabs/typewriter';
 import * as naming from '../naming';
-import { namespaceFromResource, referenceInterfaceName, referenceInterfaceAttributeName, referencePropertyName, typeAliasPrefixFromResource } from '../naming';
+import { CDK_INTERFACES } from './cdk';
 import { ResourceReference } from './reference-props';
 import { log } from '../util';
-import { SelectiveImport } from './service-submodule';
-
-/**
- * We currently disable the relationship on the properties of types because they would create a backwards incompatible change
- * by broadening the output type as types are used both in input and output. This represents:
- * Relationship counts:
- *   Resource-level (non-nested): 598
- *   Type-level (nested):         483 <- these are disabled by this flag
- *   Total:                       1081
- * Properties with relationships:
- *   Resource-level (non-nested): 493
- *   Type-level (nested):         358
- *   Total:                       851
- */
-export const GENERATE_RELATIONSHIPS_ON_TYPES = false;
 
 /**
  * Represents a cross-service property relationship that enables references
  * between resources from different AWS services.
  */
 export interface Relationship {
   /** The TypeScript interface type that provides the reference (e.g. "IRoleRef") */
-  readonly referenceType: string;
+  readonly refType: Type;
+  /** Human friendly name of the reference type for error generation (e.g. "iam.IRoleRef") */
+  readonly refTypeDisplayName: string;
   /** The property name on the reference interface that holds the reference object (e.g. "roleRef") */
-  readonly referenceName: string;
+  readonly refPropStructName: string;
   /** The property to extract from the reference object (e.g. "roleArn") */
-  readonly propName: string;
-  /** Human friendly name of the reference type for error generation (e.g. "iam.IRoleRef") */
-  readonly typeDisplayName: string;
+  readonly refPropName: string;
+}
+
+export interface RelationshipDeciderProps {
+  /**
+   * Render relationships
+   */
+  readonly enableRelationships: boolean;
+
+  /**
+   * We currently disable the relationship on the properties of types because they would create a backwards incompatible change
```

Comment:
What is the criteria to enable these? I'm wondering if we need this flag at all right now if there is no short-term plan to enable.